File: frontend/src/store/api/result/dto/forum/PostResponseOutputDTO.ts
import { PostResponseLikeDTO } from "./PostResponseLikeDTO";
import { UserDTO } from "./UserDTO";

export interface PostResponseOutputDTO {
  id: number;
  content: string;
  creationTimestamp: string;
  modificationTimestamp: string;
  post: number;
  user: UserDTO;
  likes: PostResponseLikeDTO[];
  likesAmount: number;
}

File: backend/forum-client/src/main/java/dto/response/PostResponseOutputDTO.java
package dto.response;

import dto.UserDTO;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.Set;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
public class PostResponseOutputDTO {

    private Long id;
    private String content;
    private LocalDateTime creationTimestamp;
    private LocalDateTime modificationTimestamp;
    private Long post;
    private UserDTO user;
    private Set<PostResponseLikeDTO> likes;
    private Long likesAmount;
}
